### ISSUE TITLE
Issues running on Windows

### ISSUE THREAD

**Issue by brettcannon** [python/benchmarks#5]
E.g. on Windows the binary is named `python.exe` -- same on macOS -- and the binary is stored in a `Scripts` directory instead of `bin` (#2 tries to solve this). `os.execv()` also doesn't quite operate the way you think on (at least) Windows 10 as it immediately returns to PowerShell/cmd.exe and the process runs in the background of the prompt, seeming to pause on I/O unless the user interacts with the terminal, e.g. hitting enter  (if you type `exit` at the prompt then the new process takes over the terminal window, although if you kill *that* process then you kill the entire terminal window instead of returning, including the end of the benchmark run).

**Comment by haypo** [comment]
For os.execv(): it's a quick hack to enhance the usability of the bench.py script, it avoids to have to enter manually the virtual environment.

Do you know if https://virtualenvwrapper.readthedocs.io/ or https://pypi.python.org/pypi/vex works on Windows? If yes, we should check how they work.

**Comment by brettcannon** [comment]
I don't know if they work but I can give them a try. Otherwise we should consider exposing a function in `venv` in the stdlib to figure this sort of thing out.

**Comment by ncoghlan** [comment]
I like vex for this kind of problem, as it keeps things simple: it just starts a subshell with the venv activated.

The main subprocess launching is in https://github.com/sashahart/vex/blob/master/vex/run.py

For invocation "vex <benchvenv-name> perf ...." should do the trick.

**Comment by haypo** [comment]
The commit 4f74a9df1948c6fa2f04acb69ab0ceaf4a48472a fixed the most obvious issue, fix path to the Python program inside the virtual environment.

**Comment by haypo** [comment]
You should be able to workaround the issue by creating manually the virtual environment and then using the internal --inside-venv option. Something like:

```
virtualenv env
env/bin/python -m pip install "setuptools>=18.5" "pip>=6.0"
env/bin/python -m pip install -r performance/requirements.txt 
env/bin/python -m performance run --inside-venv (...)
```

**Comment by ammaraskar** [comment]
>os.execv() also doesn't quite operate the way you think on (at least) Windows 10 as it immediately returns to PowerShell/cmd.exe and the process runs in the background of the prompt, seeming to pause on I/O unless the user interacts with the terminal

This was getting fairly annoying for me so I've opened up #8 as a quick fix for it until you guys evaluate using vex/venvwrapper

**Comment by haypo** [comment]
I tested the development version of performance: I had to install virtualenv (python3 -m pip install virtualenv). But C:\Python35\Scripts is not in the PATH by default, so it still failed. After setting PATH to include C:\Python35\Scripts, I successfully run benchmarks on Windows.

Maybe performance can be smarter and tries "python3 -m virtualenv ..." when "virtualenv" command is not in the PATH, but that's a different issue (unrelated to Windows).
